python/addons/embed_bert_pytorch.py

```diff
@@ -132,3 +413,218 @@ def __init__(self, name, **kwargs):
     def get_output(self, all_layers, pooled):
         return pooled
 
+
+class IdentityTransducer(nn.Module):
+    """A module that just passes tensor but also has an output shape."""
+    def __init__(self, input_sz):
+        super().__init__()
+        self.output_dim = input_sz
+
+    def forward(self, inputs):
+        inputs, _ = tensor_and_lengths(inputs)
+        return inputs
+
+
+def select_heads(outputs, head_idx, dim=1):
+    """Select the heads of wordpiece spans.
+
+    :param outputs: The Bert logits, Tensor[B, T, H]
+    :param head_idx: The head indices, Tensor[B, T2]
+    :param dim: The Time dimension.
+    :returns: The logits for the heads, Tensor[B, T2, H]
+    """
+    idx = head_idx.unsqueeze(-1).expand(list(head_idx.shape) + [outputs.size(-1)])
+    return torch.gather(outputs, dim, idx)
+
+
+@register_model(task='tagger', name='bert')
+class BertTaggerModel(TaggerModelBase):
+
+    def init_encoder(self, input_sz, **kwargs):
+        return IdentityTransducer(input_sz)
+
+    def make_input(self, batch_dict):
+
+        example_dict = dict({})
+        lengths = torch.from_numpy(batch_dict[self.lengths_key])
+        lengths, perm_idx = lengths.sort(0, descending=True)
+
+        if self.gpu:
+            lengths = lengths.cuda()
+        example_dict['lengths'] = lengths
+        # The vectorizer outputs a [2, T] tensor which is stacked into [B, 2, T].
+        # The first index in the 2 dim is the tokens and the second is the heads
+        for key in self.embeddings.keys():
+            example_dict[key] = self.input_tensor(key, batch_dict, perm_idx)
+
+        # This assumes all heads are aligned
+        heads = [k for k in batch_dict.keys() if k.endswith('_heads')][0]
```

Comment:
Doing thia search everytime is waste full. Can the name if the key be cached or passed in?